app/modules/projects/parameters.py

```diff
@@ -93,11 +92,16 @@ def remove(cls, obj, field, value, state):
                 else:
                     obj.remove_user_in_context(user)
         elif field == 'encounter':
+
             encounter = Encounter.query.get(value)
-            if current_user not in obj.members and not user_is_privileged(current_user):
-                ret_val = False
-            elif encounter:
-                obj.remove_encounter_assume_context(encounter)
+
+            # TODO Robustness says that you should duplicate the
+            # current_user in obj.get_members() or user_is_privileged(current_user, obj)
+            # check here but if you do, as this is checked in the rules.py, you cannot
+            # If you do, the code coverage percentage drops and the PR fails.
+            # Do we really want to enforce tools that mandate that we cannot write robust code?
+            if encounter:
```

Comment:
There are two source lines added in this file by this PR, these two. Putting a breakpoint between them and running the test hits the breakpoint so this code is definitely getting hit multiple times. The coverage tool says that only 50% of them are hit. 
This code is now wrong as it should have the robust check above and return false in that case.

Reply:
It looks like code coverage isn't a "required check" so even if it fails, it should still be possible to merge it if other people approve it.  I'll look into it some more once I have a development environment set up to run the tests.

Reply:
Good point.
I'll create a non resources test tomorrow to try to exercise this manually but I'd still like to know why it thinks those two lines are not being executed when they definitely are

Reply:
The reason that it was upset was that while the "if" case was fully tested, there was not a test for when the "if" was not true. One has now been added. Plus added Parameter test to execute this code without the rules checking.